### ISSUE TITLE
gddo-server: unused -http argument

### ISSUE THREAD

**Issue by jehiah** [golang/gddo#465]
`gddo-server` has a now-unused `-http` argument which was obsoleted in e0d34163bedcc270fcd9a2cf007eb50ea5397a37 but not removed. That commit moved to use `appengine.Main()` which thus relies on the `PORT` environment value instead of the command line argument.

```
  -http string
    	Listen for HTTP connections on this address. (default ":8080")
```

I use `gddo-server` for docs on a private github repo and updating to a newer version from this repo unexpectedly broke my setup.

**Comment by jehiah** [comment]
@stephenmw I'm happy to submit a patch to remove this flag if that's the ideal solution

**Comment by stephenmw** [comment]
So, my plan is actually to remove the call to appengine.Main within the next week. This came at a perfect time as it makes sure I will rethread -http. Actually, I should probably make plans to thread PORT as well considering people apparently use it.

That being said, gddo-server has undergone other changes that may cause problems. The new search system depends on App Engine Search and I am unsure how to make it fall back to the (terrible) Redis search. While I right now have higher priorities, I would be happy to review changes that can help others host gddo.

**Comment by jehiah** [comment]
well.. I guess it's good I don't use the search capability at the moment =)

Sounds like you'll cover this in planned work so thanks! I'll open up appropriate issues if I ever come across other details that related to running gddo against private github repos.

**Comment by stephenmw** [comment]
Yep, thanks for the bug report. GDDO is focused on the production godoc.org use case and as we have changed things no one has complained so at this point we have no idea what is broken ¯\_(ツ)_/¯.

We also depend on memcache now, does that matter?

Part of the work I am doing will allow GDDO to be run from anywhere although it will still depend on having a proxy to App Engine Search setup. We are also considering other GCP services to better improve our production deployment. Most of them shouldn't affect non GCP users (stackdriver integrations). One that might is I really want to remove our Redis dependency entirely and move to Cloud Datastore. At that point our Redis integration (if I keep it) will bit rot. If you know anyone else who cares, make sure they watch for changes and file bugs :-).

**Comment by jehiah** [comment]
I'm just benefiting from the hard work of @rjeczalik in #254 

I'm flexible for other dependencies; memcached is easy to provide and i'm sure i'll manage with Cloud Datastore or contribute a workaround. I won't miss redis. Thanks for the heads up.

**Comment by arnottcr** [comment]
It looks like the `appengine.Main()` change was reverted a while back and this flag is being used again: we good to close this out?
